### DISCUSSION TITLE
SMTP to Office 365

### DISCUSSION THREAD

**Discussion by k1ngf15h3r** [requarks/wiki#5010] [Help / Questions]
Hey Community,

a second issue with my brand new wiki.js is the binding to my office 365 mail account. With smtp server smtp.office365.com and Port 587 for STARTTLS I got an error about ssl wrong version number, is it possible that TLS 1.2 is not enabled in the docker image?

Glad to hear from you

**Comment by fasolato-omicron** [comment]
I have solved the issue, but I think it's a wiki bug.

The problem is with the parametersa passed to nodeMailer. Apparently with Office365 the `ciphers` parameter must be explicitly set. I solved editing the file `server\core\mail.js` and adding the correct parameter.

To deploy this I prepared a patch file, that I apply with a custom Dockerfile:

```patch
--- mail.js	2023-02-15 15:36:59.216931784 +0100
+++ mail.new.js	2023-02-15 15:37:19.891595852 +0100
@@ -14,9 +14,10 @@
         host: WIKI.config.mail.host,
         port: WIKI.config.mail.port,
         name: WIKI.config.mail.name,
-        secure: WIKI.config.mail.secure,
+        secure: false,
         tls: {
-          rejectUnauthorized: !(WIKI.config.mail.verifySSL === false)
+          rejectUnauthorized: !(WIKI.config.mail.verifySSL === false),
+          ciphers: 'ALL',
         }
       }
       if (_.get(WIKI.config, 'mail.user', '').length > 1) {
```

```Dockerfile
FROM ghcr.io/requarks/wiki:2.5

USER root
RUN apk add --no-cache patch

COPY mail.js.patch /wiki/server/core

RUN patch /wiki/server/core/mail.js < /wiki/server/core/mail.js.patch && rm /wiki/server/core/mail.js.patch
```
